packages/autocomplete-js/src/types/index.ts

```diff
@@ -52,6 +52,10 @@ export type InternalAutocompleteSource<TItem> = InternalAutocompleteCoreSource<
 type GetSources<TItem> = (
   params: GetSourcesParams<TItem>
 ) =>
+  // TODO: decide whether:
+  // 1. find out whether GetSources in core is also Promise / nonPromise and reuse it
+  // 2. export MaybePromise from -core (but it's public API then)
+  // 3. rewrite MaybePromise here
```

Comment:
the GetSources exported from autocomplete-core 

https://github.com/algolia/autocomplete.js/blob/f487df58a3232a5569eca26e2be3bca1b19c362b/packages/autocomplete-core/src/types/api.ts#L116-L118

seems to only allow promises, not straight arrays, and I'm not sure why it's different

Reply:
This `GetSources` is the internal type that converts the public `getSources` to return Promise in all cases. This makes code paths easier to deal with in the codebase.

You can then update the comment to something like:

> // TODO: reuse `MaybePromise` once we share this type coming from `autocomplete-core`.